## This Data Cleaning Process Focuses on an Airbnb Data Dataset

In [393]:
# Loading the dependencies
import pandas as pd

In [394]:
# Reading the dataset:
Airbnb = pd.read_csv("Airbnb_Open_Data.csv")

C:\Users\hp\AppData\Local\Temp\ipykernel_203692\470300290.py:2: DtypeWarning: Columns (25) have mixed types. Specify dtype option on import or set low_memory=False.
  Airbnb = pd.read_csv("Airbnb_Open_Data.csv")


In [395]:
# Loading first 5 rows of the dataset
Airbnb.head()

id                                              NAME      host id  \
0  1001254                Clean & quiet apt home by the park  80014485718   
1  1002102                             Skylit Midtown Castle  52335172823   
2  1002403               THE VILLAGE OF HARLEM....NEW YORK !  78829239556   
3  1002755                                               NaN  85098326012   
4  1003689  Entire Apt: Spacious Studio/Loft by central park  92037596077   

  host_identity_verified host name neighbourhood group neighbourhood  \
0            unconfirmed  Madaline            Brooklyn    Kensington   
1               verified     Jenna           Manhattan       Midtown   
2                    NaN     Elise           Manhattan        Harlem   
3            unconfirmed     Garry            Brooklyn  Clinton Hill   
4               verified    Lyndon           Manhattan   East Harlem   

        lat      long        country  ... service fee minimum nights  \
0  40.64749 -73.97237  United States  ...       $193            10.0   
1  40.75362 -73.98377  United States  ...        $28            30.0   
2  40.80902 -73.94190  United States  ...       $124             3.0   
3  40.68514 -73.95976  United States  ...        $74            30.0   
4  40.79851 -73.94399  United States  ...        $41            10.0   

  number of reviews last review  reviews per month review rate number  \
0               9.0  10/19/2021               0.21                4.0   
1              45.0   5/21/2022               0.38                4.0   
2               0.0         NaN                NaN                5.0   
3             270.0    7/5/2019               4.64                4.0   
4               9.0  11/19/2018               0.10                3.0   

  calculated host listings count  availability 365  \
0                            6.0             286.0   
1                            2.0             228.0   
2                            1.0             352.0   
3                            1.0             322.0   
4                            1.0             289.0   

                                         house_rules license  
0  Clean up and treat the home the way you'd like...     NaN  
1  Pet friendly but please confirm with me if the...     NaN  
2  I encourage you to use my kitchen, cooking and...     NaN  
3                                                NaN     NaN  
4  Please no smoking in the house, porch or on th...     NaN  

[5 rows x 26 columns]

In [396]:
# shape of the dataset
Airbnb.shape

(102599, 26)

In [397]:
## Checking the colums:
Airbnb.columns

Index(['id', 'NAME', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review', 'reviews per month',
       'review rate number', 'calculated host listings count',
       'availability 365', 'house_rules', 'license'],
      dtype='object')

In [398]:
# Checking the data_types:
Airbnb.dtypes

id                                  int64
NAME                               object
host id                             int64
host_identity_verified             object
host name                          object
neighbourhood group                object
neighbourhood                      object
lat                               float64
long                              float64
country                            object
country code                       object
instant_bookable                   object
cancellation_policy                object
room type                          object
Construction year                 float64
price                              object
service fee                        object
minimum nights                    float64
number of reviews                 float64
last review                        object
reviews per month                 float64
review rate number                float64
calculated host listings count    float64
availability 365                  

In [399]:
# Getting General information about the data: The gives us the total number of dataset present in each row and their datatypes
Airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102599 entries, 0 to 102598
Data columns (total 26 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   id                              102599 non-null  int64  
 1   NAME                            102349 non-null  object 
 2   host id                         102599 non-null  int64  
 3   host_identity_verified          102310 non-null  object 
 4   host name                       102193 non-null  object 
 5   neighbourhood group             102570 non-null  object 
 6   neighbourhood                   102583 non-null  object 
 7   lat                             102591 non-null  float64
 8   long                            102591 non-null  float64
 9   country                         102067 non-null  object 
 10  country code                    102468 non-null  object 
 11  instant_bookable                102494 non-null  object 
 12  cancellation_pol

In [400]:
# Checking for the empty sessions of the data:
# Starting with license

In [401]:
Airbnb["license"].value_counts()

license
41662/AL    2
Name: count, dtype: int64

### We have 2 non-null dataset in the row

In [402]:
# Checking for NaN values
Airbnb.license.isna().sum()

102597

### We have 102597 empty rows in the Airbnb dataset for license

## Cleaning Transformations to apply to the dataset

In [403]:
""" Deleting redundant columns.
Renaming the columns.
Dropping duplicates.
Remove the NaN values from the dataset.
Cleaning individual columns.
Checking for some more transformations."""

' Deleting redundant columns.\nRenaming the columns.\nDropping duplicates.\nRemove the NaN values from the dataset.\nCleaning individual columns.\nChecking for some more transformations.'

### Deleting redundant columns:

In [404]:
# Loading the columns
Airbnb.columns

Index(['id', 'NAME', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review', 'reviews per month',
       'review rate number', 'calculated host listings count',
       'availability 365', 'house_rules', 'license'],
      dtype='object')

In [405]:
# We keep all columns untill last review and drop the rest
cols_to_drop = ['reviews per month',
       'review rate number', 'calculated host listings count',
       'availability 365', 'house_rules', 'license', 'id']

In [406]:
cols_to_keep  = ['NAME', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review']

In [407]:
# Checking len of both columns: 7 columns
len(cols_to_drop)

7

In [408]:
# columns to keep = 19
len(cols_to_keep)

19

In [409]:
# dropping columns:
Airbnb.drop(columns = cols_to_drop, axis = 1, inplace = True)

In [410]:
Airbnb.head() # lodding 5 rows of remaining data

NAME      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
2               THE VILLAGE OF HARLEM....NEW YORK !  78829239556   
3                                               NaN  85098326012   
4  Entire Apt: Spacious Studio/Loft by central park  92037596077   

  host_identity_verified host name neighbourhood group neighbourhood  \
0            unconfirmed  Madaline            Brooklyn    Kensington   
1               verified     Jenna           Manhattan       Midtown   
2                    NaN     Elise           Manhattan        Harlem   
3            unconfirmed     Garry            Brooklyn  Clinton Hill   
4               verified    Lyndon           Manhattan   East Harlem   

        lat      long        country country code instant_bookable  \
0  40.64749 -73.97237  United States           US            False   
1  40.75362 -73.98377  United States           US            False   
2  40.80902 -73.94190  United States           US             True   
3  40.68514 -73.95976  United States           US             True   
4  40.79851 -73.94399  United States           US            False   

  cancellation_policy        room type  Construction year  price service fee  \
0              strict     Private room             2020.0  $966        $193    
1            moderate  Entire home/apt             2007.0  $142         $28    
2            flexible     Private room             2005.0  $620        $124    
3            moderate  Entire home/apt             2005.0  $368         $74    
4            moderate  Entire home/apt             2009.0  $204         $41    

   minimum nights  number of reviews last review  
0            10.0                9.0  10/19/2021  
1            30.0               45.0   5/21/2022  
2             3.0                0.0         NaN  
3            30.0              270.0    7/5/2019  
4            10.0                9.0  11/19/2018

In [411]:
# New shape:
Airbnb.shape

(102599, 19)

### Renaming the columns:

In [412]:
Airbnb.columns

Index(['NAME', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review'],
      dtype='object')

In [413]:
Airbnb.rename({"NAME":"name"}, axis = 1, inplace = True)

In [414]:
Airbnb.head()

name      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
2               THE VILLAGE OF HARLEM....NEW YORK !  78829239556   
3                                               NaN  85098326012   
4  Entire Apt: Spacious Studio/Loft by central park  92037596077   

  host_identity_verified host name neighbourhood group neighbourhood  \
0            unconfirmed  Madaline            Brooklyn    Kensington   
1               verified     Jenna           Manhattan       Midtown   
2                    NaN     Elise           Manhattan        Harlem   
3            unconfirmed     Garry            Brooklyn  Clinton Hill   
4               verified    Lyndon           Manhattan   East Harlem   

        lat      long        country country code instant_bookable  \
0  40.64749 -73.97237  United States           US            False   
1  40.75362 -73.98377  United States           US            False   
2  40.80902 -73.94190  United States           US             True   
3  40.68514 -73.95976  United States           US             True   
4  40.79851 -73.94399  United States           US            False   

  cancellation_policy        room type  Construction year  price service fee  \
0              strict     Private room             2020.0  $966        $193    
1            moderate  Entire home/apt             2007.0  $142         $28    
2            flexible     Private room             2005.0  $620        $124    
3            moderate  Entire home/apt             2005.0  $368         $74    
4            moderate  Entire home/apt             2009.0  $204         $41    

   minimum nights  number of reviews last review  
0            10.0                9.0  10/19/2021  
1            30.0               45.0   5/21/2022  
2             3.0                0.0         NaN  
3            30.0              270.0    7/5/2019  
4            10.0                9.0  11/19/2018

### Dropping duplicates.

In [415]:
# Checking for duplicates:
Airbnb.loc[Airbnb.duplicated(keep = False)]

name      host id  \
9098       Upper East Side 2 bedroom- close to Hospitals-  65193709566   
9099    Close to East Side Hospitals- Modern 2 Bedroom...  31072202372   
9100    ACADIA Spacious 2 Bedroom Apt - Close to Hospi...  95854111798   
9101      *ENCHANTMENT* Upper East Side 2 bedroom- Sunny!  73401481508   
9102          *JAMES* Amazing Spacious 2 Bedroom- Bright!  37678424985   
...                                                   ...          ...   
102594                         Spare room in Williamsburg  12312296767   
102595                      Best Location near Columbia U  77864383453   
102596                     Comfy, bright room in Brooklyn  69050334417   
102597                   Big Studio-One Stop from Midtown  11160591270   
102598                               585 sf Luxury Studio  68170633372   

       host_identity_verified    host name neighbourhood group  \
9098                 verified      Juliana           Manhattan   
9099                 verified      Juliana           Manhattan   
9100                 verified      Juliana           Manhattan   
9101              unconfirmed      Juliana           Manhattan   
9102                 verified      Juliana           Manhattan   
...                       ...          ...                 ...   
102594               verified         Krik            Brooklyn   
102595            unconfirmed        Mifan           Manhattan   
102596            unconfirmed        Megan            Brooklyn   
102597            unconfirmed  Christopher              Queens   
102598            unconfirmed      Rebecca           Manhattan   

              neighbourhood       lat      long        country country code  \
9098        Upper East Side  40.76222 -73.96030  United States           US   
9099        Upper East Side  40.76249 -73.96217  United States           US   
9100        Upper East Side  40.76021 -73.96157  United States           US   
9101        Upper East Side  40.76244 -73.96031  United States           US   
9102        Upper East Side  40.76035 -73.96133  United States           US   
...                     ...       ...       ...            ...          ...   
102594         Williamsburg  40.70862 -73.94651  United States           US   
102595  Morningside Heights  40.80460 -73.96545  United States           US   
102596           Park Slope  40.67505 -73.98045  United States           US   
102597     Long Island City  40.74989 -73.93777  United States           US   
102598      Upper West Side  40.76807 -73.98342  United States           US   

       instant_bookable cancellation_policy        room type  \
9098              False            moderate  Entire home/apt   
9099              False            moderate  Entire home/apt   
9100              False            moderate  Entire home/apt   
9101               True            moderate  Entire home/apt   
9102              False            flexible  Entire home/apt   
...                 ...                 ...              ...   
102594            False            flexible     Private room   
102595             True            moderate     Private room   
102596             True            moderate     Private room   
102597             True              strict  Entire home/apt   
102598            False            flexible  Entire home/apt   

        Construction year    price service fee  minimum nights  \
9098               2008.0    $105         $21             30.0   
9099               2008.0    $285         $57             30.0   
9100               2014.0    $586        $117             30.0   
9101               2006.0    $539        $108             30.0   
9102               2021.0    $806        $161             30.0   
...                   ...      ...         ...             ...   
102594             2003.0    $844        $169              1.0   
102595             2016.0    $837        $167              1.0   
102596             2009.0    $988        $198              

In [416]:
# Checking for the number of duplicates and non-duplicates
Airbnb.duplicated(keep = 'first').value_counts()

False    102058
True        541
Name: count, dtype: int64

In [417]:
# Duplicates: 541
# Non-duplicates: 102058

In [418]:
# Dropping off duplicates
Airbnb.drop_duplicates(keep = 'first', inplace = True)

In [419]:
Airbnb.loc[Airbnb.duplicated()] # Checks for duplicates

Empty DataFrame
Columns: [name, host id, host_identity_verified, host name, neighbourhood group, neighbourhood, lat, long, country, country code, instant_bookable, cancellation_policy, room type, Construction year, price, service fee, minimum nights, number of reviews, last review]
Index: []

### All duplicates are removed finally

### Removing NaN Values:

In [420]:
# Checking for all NaN values:
Airbnb.isna().sum().sort_values()

host id                       0
room type                     0
lat                           8
long                          8
neighbourhood                16
neighbourhood group          29
cancellation_policy          76
instant_bookable            105
country code                131
number of reviews           183
Construction year           214
price                       247
name                        250
service fee                 273
host_identity_verified      289
minimum nights              400
host name                   404
country                     532
last review               15832
dtype: int64

In [421]:
# Checking for Actual size of each feature column before performing NaN removal
Airbnb.info()

<class 'pandas.core.frame.DataFrame'>
Index: 102058 entries, 0 to 102057
Data columns (total 19 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   name                    101808 non-null  object 
 1   host id                 102058 non-null  int64  
 2   host_identity_verified  101769 non-null  object 
 3   host name               101654 non-null  object 
 4   neighbourhood group     102029 non-null  object 
 5   neighbourhood           102042 non-null  object 
 6   lat                     102050 non-null  float64
 7   long                    102050 non-null  float64
 8   country                 101526 non-null  object 
 9   country code            101927 non-null  object 
 10  instant_bookable        101953 non-null  object 
 11  cancellation_policy     101982 non-null  object 
 12  room type               102058 non-null  object 
 13  Construction year       101844 non-null  float64
 14  price                   1

In [422]:
Airbnb.name.shape

(102058,)

In [423]:
Airbnb.name.isna().sum()

250

In [424]:
## First dropping off the last review
Airbnb.drop(columns = ["last review"], inplace = True, axis = 1)

In [425]:
Airbnb.isna().sum().sort_values()

host id                     0
room type                   0
lat                         8
long                        8
neighbourhood              16
neighbourhood group        29
cancellation_policy        76
instant_bookable          105
country code              131
number of reviews         183
Construction year         214
price                     247
name                      250
service fee               273
host_identity_verified    289
minimum nights            400
host name                 404
country                   532
dtype: int64

In [426]:
## Dropping off some NaN values:
Airbnb.dropna(how ='any', inplace = True, axis = 0)

In [427]:
Airbnb.isna().sum()

name                      0
host id                   0
host_identity_verified    0
host name                 0
neighbourhood group       0
neighbourhood             0
lat                       0
long                      0
country                   0
country code              0
instant_bookable          0
cancellation_policy       0
room type                 0
Construction year         0
price                     0
service fee               0
minimum nights            0
number of reviews         0
dtype: int64

### Cleaning individual columns.

In [428]:
# Checking the columns
Airbnb.columns

Index(['name', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews'],
      dtype='object')

In [429]:
Airbnb.head()

name      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            unconfirmed  Madaline            Brooklyn          Kensington   
1               verified     Jenna           Manhattan             Midtown   
4               verified    Lyndon           Manhattan         East Harlem   
5               verified  Michelle           Manhattan         Murray Hill   
7            unconfirmed      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code instant_bookable  \
0  40.64749 -73.97237  United States           US            False   
1  40.75362 -73.98377  United States           US            False   
4  40.79851 -73.94399  United States           US            False   
5  40.74767 -73.97500  United States           US             True   
7  40.68688 -73.95596  United States           US            False   

  cancellation_policy        room type  Construction year    price  \
0              strict     Private room             2020.0    $966    
1            moderate  Entire home/apt             2007.0    $142    
4            moderate  Entire home/apt             2009.0    $204    
5            flexible  Entire home/apt             2013.0    $577    
7            moderate     Private room             2009.0  $1,060    

  service fee  minimum nights  number of reviews  
0       $193             10.0                9.0  
1        $28             30.0               45.0  
4        $41             10.0                9.0  
5       $115              3.0               74.0  
7       $212             45.0               49.0

In [430]:
# 1. Making the column: "host_identity_verified"
Airbnb.host_identity_verified

0         unconfirmed
1            verified
4            verified
5            verified
7         unconfirmed
             ...     
102029    unconfirmed
102030       verified
102031       verified
102032    unconfirmed
102040       verified
Name: host_identity_verified, Length: 99342, dtype: object

In [431]:
# Changing all to upper_case 
Airbnb.host_identity_verified.str.upper()

0         UNCONFIRMED
1            VERIFIED
4            VERIFIED
5            VERIFIED
7         UNCONFIRMED
             ...     
102029    UNCONFIRMED
102030       VERIFIED
102031       VERIFIED
102032    UNCONFIRMED
102040       VERIFIED
Name: host_identity_verified, Length: 99342, dtype: object

In [432]:
Airbnb["host_identity_verified"] = Airbnb["host_identity_verified"].str.upper()

In [433]:
Airbnb.head() # Column changed to uppercase

name      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
4               VERIFIED    Lyndon           Manhattan         East Harlem   
5               VERIFIED  Michelle           Manhattan         Murray Hill   
7            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code instant_bookable  \
0  40.64749 -73.97237  United States           US            False   
1  40.75362 -73.98377  United States           US            False   
4  40.79851 -73.94399  United States           US            False   
5  40.74767 -73.97500  United States           US             True   
7  40.68688 -73.95596  United States           US            False   

  cancellation_policy        room type  Construction year    price  \
0              strict     Private room             2020.0    $966    
1            moderate  Entire home/apt             2007.0    $142    
4            moderate  Entire home/apt             2009.0    $204    
5            flexible  Entire home/apt             2013.0    $577    
7            moderate     Private room             2009.0  $1,060    

  service fee  minimum nights  number of reviews  
0       $193             10.0                9.0  
1        $28             30.0               45.0  
4        $41             10.0                9.0  
5       $115              3.0               74.0  
7       $212             45.0               49.0

### Renaming feature: instant_bookable to boolean

In [434]:
Airbnb["instant_bookable"].value_counts()

instant_bookable
False    49878
True     49464
Name: count, dtype: int64

In [435]:
# Changing from object - boolean
Airbnb["instant_bookable"] = Airbnb["instant_bookable"].astype("boolean")

In [436]:
Airbnb.instant_bookable.dtype

BooleanDtype

### Changing all False as 0 and True as 1

In [437]:
Airbnb["instant_bookable"] = Airbnb["instant_bookable"].apply(lambda x: 1 if x == True else 0)

In [438]:
Airbnb.head()

name      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
4               VERIFIED    Lyndon           Manhattan         East Harlem   
5               VERIFIED  Michelle           Manhattan         Murray Hill   
7            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
4  40.79851 -73.94399  United States           US                 0   
5  40.74767 -73.97500  United States           US                 1   
7  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year    price  \
0              strict     Private room             2020.0    $966    
1            moderate  Entire home/apt             2007.0    $142    
4            moderate  Entire home/apt             2009.0    $204    
5            flexible  Entire home/apt             2013.0    $577    
7            moderate     Private room             2009.0  $1,060    

  service fee  minimum nights  number of reviews  
0       $193             10.0                9.0  
1        $28             30.0               45.0  
4        $41             10.0                9.0  
5       $115              3.0               74.0  
7       $212             45.0               49.0

### Checking for number of number of instant_bookable and not instant_bookable

In [439]:
Airbnb["instant_bookable"].value_counts()

instant_bookable
0    49878
1    49464
Name: count, dtype: int64

### This implies we have: No istant_bookable of 49878 and instant_bookable as 49464 

In [440]:
Airbnb.head()

name      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
4               VERIFIED    Lyndon           Manhattan         East Harlem   
5               VERIFIED  Michelle           Manhattan         Murray Hill   
7            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
4  40.79851 -73.94399  United States           US                 0   
5  40.74767 -73.97500  United States           US                 1   
7  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year    price  \
0              strict     Private room             2020.0    $966    
1            moderate  Entire home/apt             2007.0    $142    
4            moderate  Entire home/apt             2009.0    $204    
5            flexible  Entire home/apt             2013.0    $577    
7            moderate     Private room             2009.0  $1,060    

  service fee  minimum nights  number of reviews  
0       $193             10.0                9.0  
1        $28             30.0               45.0  
4        $41             10.0                9.0  
5       $115              3.0               74.0  
7       $212             45.0               49.0

In [441]:
# Resetting the index of the dataset:
Airbnb.reset_index(inplace = True)

In [442]:
Airbnb.head()

index                                              name      host id  \
0      0                Clean & quiet apt home by the park  80014485718   
1      1                             Skylit Midtown Castle  52335172823   
2      4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
3      5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
4      7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
2               VERIFIED    Lyndon           Manhattan         East Harlem   
3               VERIFIED  Michelle           Manhattan         Murray Hill   
4            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
2  40.79851 -73.94399  United States           US                 0   
3  40.74767 -73.97500  United States           US                 1   
4  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year    price  \
0              strict     Private room             2020.0    $966    
1            moderate  Entire home/apt             2007.0    $142    
2            moderate  Entire home/apt             2009.0    $204    
3            flexible  Entire home/apt             2013.0    $577    
4            moderate     Private room             2009.0  $1,060    

  service fee  minimum nights  number of reviews  
0       $193             10.0                9.0  
1        $28             30.0               45.0  
2        $41             10.0                9.0  
3       $115              3.0               74.0  
4       $212             45.0               49.0

In [443]:
# Checking the datatype of the price feature:
Airbnb["price"].dtype

dtype('O')

In [444]:
Airbnb["price"] = Airbnb["price"].str.replace("$", "")

In [445]:
Airbnb["price"] = Airbnb["price"].str.replace(",", "")

In [446]:
Airbnb["price"] = Airbnb["price"].str.strip()

In [451]:
# Changing to int:
Airbnb["price"]

0         966
1         142
2         204
3         577
4        1060
         ... 
99337     822
99338     455
99339    1078
99340     103
99341     982
Name: price, Length: 99342, dtype: int32

In [448]:
# Applying same thing for service fees

In [452]:
Airbnb["service fee"] = Airbnb["service fee"].str.replace("$", "")

In [453]:
Airbnb["service fee"] = Airbnb["service fee"].str.replace(",", "")

In [454]:
Airbnb["service fee"] = Airbnb["service fee"].str.strip()

In [455]:
Airbnb["service fee"]

0        193
1         28
2         41
3        115
4        212
        ... 
99337    164
99338     91
99339    216
99340     21
99341    196
Name: service fee, Length: 99342, dtype: object

In [456]:
Airbnb.head()

index                                              name      host id  \
0      0                Clean & quiet apt home by the park  80014485718   
1      1                             Skylit Midtown Castle  52335172823   
2      4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
3      5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
4      7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
2               VERIFIED    Lyndon           Manhattan         East Harlem   
3               VERIFIED  Michelle           Manhattan         Murray Hill   
4            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
2  40.79851 -73.94399  United States           US                 0   
3  40.74767 -73.97500  United States           US                 1   
4  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year  price service fee  \
0              strict     Private room             2020.0    966         193   
1            moderate  Entire home/apt             2007.0    142          28   
2            moderate  Entire home/apt             2009.0    204          41   
3            flexible  Entire home/apt             2013.0    577         115   
4            moderate     Private room             2009.0   1060         212   

   minimum nights  number of reviews  
0            10.0                9.0  
1            30.0               45.0  
2            10.0                9.0  
3             3.0               74.0  
4            45.0               49.0

In [462]:
Airbnb["price"].dtype

dtype('O')

In [463]:
Airbnb["service fee"].dtype

dtype('O')

In [464]:
Airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99342 entries, 0 to 99341
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   index                   99342 non-null  int64  
 1   name                    99342 non-null  object 
 2   host id                 99342 non-null  int64  
 3   host_identity_verified  99342 non-null  object 
 4   host name               99342 non-null  object 
 5   neighbourhood group     99342 non-null  object 
 6   neighbourhood           99342 non-null  object 
 7   lat                     99342 non-null  float64
 8   long                    99342 non-null  float64
 9   country                 99342 non-null  object 
 10  country code            99342 non-null  object 
 11  instant_bookable        99342 non-null  int64  
 12  cancellation_policy     99342 non-null  object 
 13  room type               99342 non-null  object 
 14  Construction year       99342 non-null

In [465]:
Airbnb = Airbnb.astype({"price": "int", "service fee": "int"})

In [467]:
Airbnb.dtypes

index                       int64
name                       object
host id                     int64
host_identity_verified     object
host name                  object
neighbourhood group        object
neighbourhood              object
lat                       float64
long                      float64
country                    object
country code               object
instant_bookable            int64
cancellation_policy        object
room type                  object
Construction year         float64
price                       int32
service fee                 int32
minimum nights            float64
number of reviews         float64
dtype: object

### Finally cleaned data with 5 rows

In [469]:
Airbnb.head()

index                                              name      host id  \
0      0                Clean & quiet apt home by the park  80014485718   
1      1                             Skylit Midtown Castle  52335172823   
2      4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
3      5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
4      7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
2               VERIFIED    Lyndon           Manhattan         East Harlem   
3               VERIFIED  Michelle           Manhattan         Murray Hill   
4            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
2  40.79851 -73.94399  United States           US                 0   
3  40.74767 -73.97500  United States           US                 1   
4  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year  price  service fee  \
0              strict     Private room             2020.0    966          193   
1            moderate  Entire home/apt             2007.0    142           28   
2            moderate  Entire home/apt             2009.0    204           41   
3            flexible  Entire home/apt             2013.0    577          115   
4            moderate     Private room             2009.0   1060          212   

   minimum nights  number of reviews  
0            10.0                9.0  
1            30.0               45.0  
2            10.0                9.0  
3             3.0               74.0  
4            45.0               49.0

### Saving Cleaned dataset

In [476]:
Airbnb.to_csv("Cleaned_Airbnb.csv", index = False)

In [477]:
# Reading Cleaned dataset
Airbnb_cleaned = pd.read_csv("Cleaned_Airbnb.csv")

In [478]:
Airbnb_cleaned.head()

index                                              name      host id  \
0      0                Clean & quiet apt home by the park  80014485718   
1      1                             Skylit Midtown Castle  52335172823   
2      4  Entire Apt: Spacious Studio/Loft by central park  92037596077   
3      5         Large Cozy 1 BR Apartment In Midtown East  45498551794   
4      7                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
2               VERIFIED    Lyndon           Manhattan         East Harlem   
3               VERIFIED  Michelle           Manhattan         Murray Hill   
4            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
2  40.79851 -73.94399  United States           US                 0   
3  40.74767 -73.97500  United States           US                 1   
4  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year  price  service fee  \
0              strict     Private room             2020.0    966          193   
1            moderate  Entire home/apt             2007.0    142           28   
2            moderate  Entire home/apt             2009.0    204           41   
3            flexible  Entire home/apt             2013.0    577          115   
4            moderate     Private room             2009.0   1060          212   

   minimum nights  number of reviews  
0            10.0                9.0  
1            30.0               45.0  
2            10.0                9.0  
3             3.0               74.0  
4            45.0               49.0

In [480]:
Airbnb_cleaned.drop(columns = ["index"], inplace = True, axis = 1)

In [481]:
Airbnb_cleaned.head()

name      host id  \
0                Clean & quiet apt home by the park  80014485718   
1                             Skylit Midtown Castle  52335172823   
2  Entire Apt: Spacious Studio/Loft by central park  92037596077   
3         Large Cozy 1 BR Apartment In Midtown East  45498551794   
4                                   BlissArtsSpace!  90821839709   

  host_identity_verified host name neighbourhood group       neighbourhood  \
0            UNCONFIRMED  Madaline            Brooklyn          Kensington   
1               VERIFIED     Jenna           Manhattan             Midtown   
2               VERIFIED    Lyndon           Manhattan         East Harlem   
3               VERIFIED  Michelle           Manhattan         Murray Hill   
4            UNCONFIRMED      Emma            Brooklyn  Bedford-Stuyvesant   

        lat      long        country country code  instant_bookable  \
0  40.64749 -73.97237  United States           US                 0   
1  40.75362 -73.98377  United States           US                 0   
2  40.79851 -73.94399  United States           US                 0   
3  40.74767 -73.97500  United States           US                 1   
4  40.68688 -73.95596  United States           US                 0   

  cancellation_policy        room type  Construction year  price  service fee  \
0              strict     Private room             2020.0    966          193   
1            moderate  Entire home/apt             2007.0    142           28   
2            moderate  Entire home/apt             2009.0    204           41   
3            flexible  Entire home/apt             2013.0    577          115   
4            moderate     Private room             2009.0   1060          212   

   minimum nights  number of reviews  
0            10.0                9.0  
1            30.0               45.0  
2            10.0                9.0  
3             3.0               74.0  
4            45.0               49.0